system_design/design_web_crawler.md

```diff
@@ -0,0 +1,228 @@
+
+# Web Crawler
+
+**Uses**
+
+Web crawlers are useful for accomplishing the following tasks.
+- How search engines get their data (ex. Google, Baidu)
+- Test web page links
+- Monitor content or structure updates on a page
+- Mirroring 
+    - mirroring: making a copy of a website
+- Copyright infrigement checks
```

Comment:
option: Also great for creating APIs for websites that don't already have one.

Reply:
Ooh. Tell me more!